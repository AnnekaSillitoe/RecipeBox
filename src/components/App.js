import React, {Component, Fragment} from "react";
import "../css/index.css";
import AddRecipe from "./AddRecipe";
import DisplayRecipes from "./DisplayRecipes";
import DisplayRecipe from "./DisplayRecipe";

class App extends Component {
  state = {
    displayAdd: false,
    displayRecipe: null
  };

  displayRecipe = (recipeId) => {
    this.setState({
      displayRecipe: recipeId,
    })
  };

  redirectBack = (location) => {
    const display = location === "addRecipe" ? "displayAdd" : "displayRecipe";
    this.setState({
      [display]: false
    });
  };

  render() {
    const { displayAdd, displayRecipe } = this.state;
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Recipes</h1>
        </header>
        {!displayRecipe &&
        <Fragment>
        {!displayAdd && (
          <div className="mt-5">
            <button className="btn btn-dark float-right mr-3" onClick={() => this.setState({ displayAdd: true })}>
              Add recipe
            </button>
          </div>
        )}
        {displayAdd && <AddRecipe redirectBack={this.redirectBack}/>}
        {!displayAdd && <DisplayRecipes displayRecipe={this.displayRecipe} />}
        </Fragment>}
        {typeof displayRecipe === "string" && <DisplayRecipe recipeId={displayRecipe} redirectBack={this.redirectBack}/>}
      </div>
    );
  }
}

export default App;
