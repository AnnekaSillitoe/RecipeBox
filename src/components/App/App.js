import React, { Component, Fragment } from "react";
import AddRecipe from "../AddRecipe";
import DisplayRecipes from "../DisplayRecipes";
import DisplayRecipe from "../DisplayRecipe";
import Header from "../Header/Header";
import ButtonPrimary from "../Buttons/ButtonPrimary/ButtonPrimary";

class App extends Component {
  state = {
    displayAdd: false,
    displayRecipe: null
  };

  displayRecipe = recipeId => {
    this.setState({
      displayRecipe: recipeId
    });
  };

  redirectBack = location => {
    const display = location === "addRecipe" ? "displayAdd" : "displayRecipe";
    this.setState({
      [display]: false
    });
  };

  displayAddRecipe = () => {
    this.setState({ displayAdd: true });
  };

  render() {
    const { displayAdd, displayRecipe } = this.state;
    return (
      <div className="App">
        <Header />
        {!displayRecipe && (
          <Fragment>
            {!displayAdd && (
              <ButtonPrimary
                buttonClasses="float-right mt-1 mr-3"
                onClick={this.displayAddRecipe}
                buttonText="Add a recipe"
              />
            )}
            {displayAdd && <AddRecipe redirectBack={this.redirectBack} />}
            {!displayAdd && (
              <DisplayRecipes displayRecipe={this.displayRecipe} />
            )}
          </Fragment>
        )}
        {typeof displayRecipe === "string" && (
          <DisplayRecipe
            recipeId={displayRecipe}
            redirectBack={this.redirectBack}
          />
        )}
      </div>
    );
  }
}

export default App;
