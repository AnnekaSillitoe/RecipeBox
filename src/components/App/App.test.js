import React from 'react';
import { shallow } from 'enzyme';
import App from "./App";

test('App renders without crashing', () => {
  const app = shallow(<App />);
  expect(app).toMatchSnapshot();
});