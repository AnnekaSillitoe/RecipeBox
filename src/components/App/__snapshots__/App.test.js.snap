// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Unknown>
          <ButtonPrimary
                    buttonClasses="float-right mt-1 mr-3"
                    buttonText="Add a recipe"
                    onClick={[Function]}
          />
          <DisplayRecipes
                    displayRecipe={[Function]}
          />
</Unknown>,
        false,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ButtonPrimary
              buttonClasses="float-right mt-1 mr-3"
              buttonText="Add a recipe"
              onClick={[Function]}
/>,
            false,
            <DisplayRecipes
              displayRecipe={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonClasses": "float-right mt-1 mr-3",
              "buttonText": "Add a recipe",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "displayRecipe": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Unknown>
            <ButtonPrimary
                        buttonClasses="float-right mt-1 mr-3"
                        buttonText="Add a recipe"
                        onClick={[Function]}
            />
            <DisplayRecipes
                        displayRecipe={[Function]}
            />
</Unknown>,
          false,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ButtonPrimary
                buttonClasses="float-right mt-1 mr-3"
                buttonText="Add a recipe"
                onClick={[Function]}
/>,
              false,
              <DisplayRecipes
                displayRecipe={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "buttonClasses": "float-right mt-1 mr-3",
                "buttonText": "Add a recipe",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "displayRecipe": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
